evaluator/evaluate_search.go

```diff
@@ -141,7 +141,24 @@ eventMatcher:
 
 func (rule *RuleEvaluator) getMatcherValues(ctx context.Context, matcher sigma.FieldMatcher) ([]string, error) {
 	matcherValues := []string{}
-	for _, value := range matcher.Values {
+	for _, abstractValue := range matcher.Values {
+		value := ""
+
+		switch abstractValue := abstractValue.(type) {
+		case string:
+			value = abstractValue
+		case int:
+			value = fmt.Sprintf("%v", abstractValue)
+		case float32:
+			value = fmt.Sprintf("%v", abstractValue)
+		case float64:
+			value = fmt.Sprintf("%v", abstractValue)
+		case bool:
+			value = fmt.Sprintf("%v", abstractValue)
+		default:
+			return nil, fmt.Errorf("expected scalar field matching value got: %v", abstractValue)
```

Comment:
Including the concrete type in the error will help debug this
```suggestion
			return nil, fmt.Errorf("expected scalar field matching value got: %v (%T)", abstractValue, abstractValue)
```